shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -244,6 +316,11 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
     * Shorthands for deriving function scopes.
     */
   def derive(name: Str, params: Ls[Str]): Scope = Scope(name, params, this)
+
+  /**
+    * An iterator over the type symbols declared in the current scope
```

Comment:
That's not an iterator. You should probably use `valuesIterator` instead, and update the return type.

Reply:
On the other hand this is not being used anywhere, so I'll remove it.